pipelines/broad/rna_seq/RNAWithUMIsPipeline.wdl

```diff
@@ -0,0 +1,329 @@
+	File metrics = rnaseqc2.metrics
+		
+  }
+}
+
+task STAR {
+	input {
+		File bam
+		File starIndex
+	}
+	Int disk_space = ceil(2.2 * size(bam, "GB") + size(starIndex, "GB")) + 250
+
+	command <<<
+		echo $(date +"[%b %d %H:%M:%S] Extracting STAR index")
+		mkdir star_index
+		tar -xvf ~{starIndex} -C star_index --strip-components=1
+
+		STAR --readFilesIn ~{bam} --readFilesType SAM PE --readFilesCommand samtools view -h \
+			--runMode alignReads --genomeDir star_index --outSAMtype BAM Unsorted --runThreadN 8 \
+			--limitSjdbInsertNsj 1200000 --outSAMstrandField intronMotif --outSAMunmapped Within \
+			--outFilterType BySJout --outFilterMultimapNmax 20 --outFilterScoreMinOverLread 0.33 \
+			--outFilterMatchNminOverLread 0.33 --outFilterMismatchNmax 999 --outFilterMismatchNoverLmax 0.1 \
+			--alignIntronMin 20 --alignIntronMax 1000000 --alignMatesGapMax 1000000 --alignSJoverhangMin 8 \
+			--alignSJDBoverhangMin 1 --alignSoftClipAtReferenceEnds Yes --chimSegmentMin 15 --chimMainSegmentMultNmax 1 \
+			--chimOutType WithinBAM SoftClip --chimOutJunctionFormat 0 --twopassMode Basic --quantMode TranscriptomeSAM --quantTranscriptomeBan Singleend
+
+	>>>
+
+	runtime {
+		docker : "us.gcr.io/tag-team-160914/neovax-tag-rnaseq:v1"
+		disks : "local-disk " + disk_space + " HDD"
+		memory : "64GB"
+		cpu : "8"
+		preemptible: 0
+	}
+
+	output {
+		File aligned_bam = "Aligned.out.bam"
+		File transcriptome_bam = "Aligned.toTranscriptome.out.bam"
+	}
+}
+
+task ExtractUMIs {
+	input {
+		File bam
+		String read1Structure
+		String read2Structure
+	}
+
+	Int disk_space = ceil(2.2 * size(bam, "GB")) + 50
+
+	command <<<
+		fgbio ExtractUmisFromBam --input ~{bam} \
+			--read-structure ~{read1Structure} \
+			--read-structure ~{read2Structure} \
+			--molecular-index-tags RX \
+			--output extractUMIs.out.bam
+	>>>
+
+	runtime {
+		docker : "quay.io/biocontainers/fgbio@sha256:a8e5cf58c318bffba3b2b694a3640ecd9e8106cee2e33b75710c0e8215138b6e"
+		disks : "local-disk " + disk_space + " HDD"
+		preemptible: 0
+	}
+
+	output {
+		File bam_umis_extracted = "extractUMIs.out.bam"
+	}
+}
+
+task rnaseqc2 {
+	input {
+		File bam_file
+		File genes_gtf
+		String sample_id
+		File exon_bed = "gs://gtex-resources/GENCODE/gencode.v26.GRCh38.insert_size_intervals_geq1000bp.bed"
+	}
+	
+	Int disk_space = ceil(size(bam_file, 'GB') + size(genes_gtf, 'GB')) + 100
+
+	command {
+		set -euo pipefail
+		echo $(date +"[%b %d %H:%M:%S] Running RNA-SeQC 2")
+		rnaseqc ~{genes_gtf} ~{bam_file} . -s ~{sample_id} -v --bed ~{exon_bed}
+		echo "  * compressing outputs"
+		gzip *.gct
+		echo $(date +"[%b %d %H:%M:%S] done")
+	}
+
+	output {
+		File gene_tpm = "${sample_id}.gene_tpm.gct.gz"
+		File gene_counts = "${sample_id}.gene_reads.gct.gz"
+		File exon_counts = "${sample_id}.exon_reads.gct.gz"
+		File insert_size_histogram = "${sample_id}.fragmentSizes.txt"
+		File metrics = "${sample_id}.metrics.tsv"
+	}
+
+	runtime {
+		docker: "us.gcr.io/broad-dsde-methods/ckachulis/rnaseqc:2.4.2"
```

Comment:
@kachulis can we get the Dockerfile for this?

Reply:
Here it is! https://github.com/getzlab/rnaseqc/blob/master/Dockerfile

Reply:
Thanks